File: node/quickstart.mjs
const API_URL = "https://api.compsniper.com/v1/scrape";
const apiKey = process.env.COMPSNIPER_API_KEY?.trim();
if (!apiKey) throw new Error("Set COMPSNIPER_API_KEY before running this example.");

const url = new URL(API_URL);
url.searchParams.set("keyword", process.env.COMPSNIPER_KEYWORD ?? "iphone 15 pro");
url.searchParams.set("count", "10");
url.searchParams.set("itemCondition", "used");

const response = await fetch(url, {
  headers: { Authorization: `Bearer ${apiKey}` },
  signal: AbortSignal.timeout(75_000),
});
const data = await response.json();
if (!response.ok) throw new Error(data.error ?? `CompSniper returned HTTP ${response.status}`);

console.log("Listings returned:", data.totalItems);
console.log("Median:", data.summary?.median, data.summary?.currency ?? "");
console.log("Realistic range:", data.summary?.p25, "-", data.summary?.p75);
console.log("Monthly requests remaining:", response.headers.get("x-usage-remaining") ?? "unknown");

for (const item of data.items.slice(0, 5)) {
  console.log(`- ${item.title} | ${item.soldPrice} ${item.soldCurrency} | ${item.endedAt}`);
}

File: python/quickstart.py
"""Fetch ten recent eBay sold listings and print the CompSniper price summary."""

from __future__ import annotations

import os

import requests

API_URL = "https://api.compsniper.com/v1/scrape"


def required_api_key() -> str:
    key = os.environ.get("COMPSNIPER_API_KEY", "").strip()
    if not key:
        raise SystemExit("Set COMPSNIPER_API_KEY before running this example.")
    return key


response = requests.get(
    API_URL,
    headers={"Authorization": f"Bearer {required_api_key()}"},
    params={
        "keyword": os.environ.get("COMPSNIPER_KEYWORD", "iphone 15 pro"),
        "count": 10,
        "itemCondition": "used",
    },
    timeout=75,
)
response.raise_for_status()
data = response.json()

summary = data.get("summary") or {}
print(f"Listings returned: {data['totalItems']}")
print(f"Median: {summary.get('median')} {summary.get('currency') or ''}".rstrip())
print(f"Realistic range: {summary.get('p25')} - {summary.get('p75')}")
print(f"Monthly requests remaining: {response.headers.get('X-Usage-Remaining', 'unknown')}")

for item in data.get("items", [])[:5]:
    print(f"- {item.get('title')} | {item.get('soldPrice')} {item.get('soldCurrency')} | {item.get('endedAt')}")
